### PR TITLE
Improve error handling in frontend

### LINKED ISSUE #36: Improve error handling in the frontend
Right now there's only one place where there is some kind of error handling, and it's a catch-all kind of alert.

We should have more error alerts and logs, and they should be more granular.

### PR DESCRIPTION
**Pull request by matiasgaratortiz**
Closes #36

### CODE REVIEW

**Comment by matiasgaratortiz** [comment]
@fvictorio @phahulin Please, if you can review it, and comment if you have any ideas.

**Comment by fvictorio** [comment]
We should also emit a proper error if the contract is not deployed in the current network. I think this error will be thrown by the `PobaContract.deployed()` function.

File: frontend/src/PoBA.js [hunk 1]
```diff
@@ -46,30 +46,41 @@ class PoBA extends Component {
...
+        whenAccounts.push(this.pobaContract.accounts(account, i))
+      }
 
-    const registeredAccounts = await Promise.all(whenAccounts)
+      const registeredAccounts = await Promise.all(whenAccounts)
 
-    this.setState({ registeredAccounts })
+      this.setState({ registeredAccounts })
+    } catch (e) {
+      console.error('There was a problem deploying the contract', e)
```
Comment (phahulin) [inline frontend/src/PoBA.js]: Let's change the error message to something like "Contract is not deployed on this network".

File: frontend/src/PoBA.js [hunk 2]
```diff
@@ -46,30 +46,41 @@ class PoBA extends Component {
 
     PobaContract.setProvider(web3.currentProvider)
 
-    this.pobaContract = await PobaContract.deployed()
+    try {
+      this.pobaContract = await PobaContract.deployed()
 
-    const registeredAcountsCount = await this.pobaContract.accountsLength.call(account)
+      const registeredAcountsCount = await this.pobaContract.accountsLength.call(account)
```
Comment (phahulin) [inline frontend/src/PoBA.js]: There's a typo in variable name, should be registeredA**cc**ountsCount

**Pull request merged**